src/asdf.lisp

```diff
@@ -28,7 +28,9 @@
                       (component-pathname component)))
              (compiled (eco.compiler:compile-template
                         parsed
-                        (template-package component))))
+                        (template-package component)))
+             ;; Need to maintain reference EQuality for uninterned symbols.
+             (*print-circle* t))
```

Comment:
Sorry, can you explain how the value of `*print-circle*` ensures objects are `EQ`?

Reply:
This had to do with uninterned symbols in the generated source code,

See also this, recently on Reddit,

https://www.reddit.com/r/Common_Lisp/comments/cl4pr0/anything_wrong_with_n_and_n_in_literal_source/

Cosmetically, <expr-tag> emitter's use of gensym may not be pretty. We could probably use %eco-stream instead, but if prettiness is not a requirement, then I'm more comfortable with gensym as a safer option in all situations (whatever the future may bring).

Reply:
:+1: that makes sense, thanks!